### PR TITLE
feat(runtime): add AtUri.parse() / parseOrNull() helpers

### LINKED ISSUE #57: Add AtUri.parse() helper returning typed (repo, collection, rkey, fragment)
## Context

`AtUri` is currently a value class wrapping a raw string with no parsing helpers. Consumers calling `com.atproto.repo.deleteRecord` (or any operation keyed off the constituent parts of an AT URI) need to split the URI into `repo`, `collection`, `rkey`, and (rarely) a fragment.

Hit while writing the like/repost data layer in nubecita: to call `deleteRecord` against a like/repost record's uri, the consumer parses

```
at://did:plc:xxxxx/app.bsky.feed.like/3lkrkey
```

into `{ repo = "did:plc:xxxxx", collection = "app.bsky.feed.like", rkey = "3lkrkey" }`. Today every consumer reinvents this — see the inline `parseAtUri` in `DefaultLikeRepostRepository`.

## Proposal

Add a parser on the `AtUri` value class (or as a top-level `runtime` helper):

```kotlin
public data class AtUriParts(
    val repo: AtIdentifier,
    val collection: Nsid?,
    val rkey: RecordKey?,
    val fragment: String?,
)

public fun AtUri.parse(): AtUriParts
public fun AtUri.parseOrNull(): AtUriParts?
```

Per the AT URI spec (https://atproto.com/specs/at-uri-scheme):

- Repo (DID or handle) is required
- Collection (NSID) is optional
- Record key is optional (only valid with collection)
- Fragment after `#` is optional (selects a sub-record)

The parser should:

- Reject inputs that don't start with `at://` (throw `IllegalArgumentException` from `parse`, return `null` from `parseOrNull`)
- Validate the repo segment is a syntactically-valid `AtIdentifier` (DID or handle)
- Return `null`/throw on empty path segments

## Why upstream

Every consumer of the lib that touches record-uri-keyed write paths (likes, reposts, follows, blocks, lists, profile edits, replies addressed by parent uri, …) hits this. Centralizing the parser in `runtime` removes duplicated string-splitting and keeps the validation rules in lockstep with the spec.

## Surface

Sibling to `Datetime` / `Cid` / etc. in `runtime/src/commonMain/kotlin/io/github/kikin81/atproto/runtime/StringFormats.kt` (or a dedicated `AtUriParser.kt`).

### PR DESCRIPTION
**Pull request by kikin81**
## Summary

- Adds `AtUri.parse()` / `AtUri.parseOrNull()` extensions returning a typed `AtUriParts(repo, collection, rkey, fragment)` so consumers stop reinventing string-splitting at every record-uri-keyed call site.
- Validation is intentionally structural-only — `at://` prefix, non-empty segments, single trailing-slash tolerance, fragment stripping. Semantic validity of the repo segment (DID/handle syntax) is left to the `Did`/`Handle`/`AtIdentifier` wrappers in a separate ticket; this parser composes for free once those validators exist.
- Fragment returned as the bare suffix (no leading `#`) matching `java.net.URI.getFragment()` conventions.
- New file `runtime/src/commonMain/.../AtUriParser.kt` with sibling `AtUriParserTest.kt` (18 cases). Did not touch `StringFormats.kt` to keep its single-responsibility character (value-class declarations only).

Closes #57. Tracks kikinlex-yzm.

## Test plan

- [x] `./gradlew :runtime:jvmTest` — all 18 new tests pass, full runtime suite green
- [x] `./gradlew :runtime:spotlessCheck` — clean
- [ ] Reviewer sanity-check on edge cases: trailing-slash normalization, double-slash rejection, fragment-without-record validity, garbage-repo accepted (structural-only by design)

## Out of scope (follow-ups)

- Spec-correct `Did`/`Handle`/`AtIdentifier` syntax validators — separate ticket.
- `AtUriParts.toAtUri(): AtUri` round-trip helper — punted; not a blocker, will file if a consumer asks.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: runtime/src/commonMain/kotlin/io/github/kikin81/atproto/runtime/AtUriParser.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    if (!raw.startsWith("at://")) return null
+    val body = raw.substring("at://".length)
+
+    val hashIdx = body.indexOf('#')
+    val pathPart = if (hashIdx >= 0) body.substring(0, hashIdx) else body
+    val fragment = if (hashIdx >= 0) body.substring(hashIdx + 1) else null
+
+    val rawSegments = pathPart.split('/')
+    // Tolerate one trailing slash after the last non-empty segment.
+    val segments = if (rawSegments.size > 1 && rawSegments.last().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (kikin81) [reply runtime/src/commonMain/kotlin/io/github/kikin81/atproto/runtime/AtUriParser.kt]: Fixed in 2b3bde1f. Took option (a) — restricted normalization to `rawSegments.size in 2..3` so `at://repo/coll/rkey/` now rejects rather than silently accepting. KDoc was already accurate; the code is now in line with it. Added a `parse_trailingSlashAfterRkey_throws` test to lock the behavior in.

File: runtime/src/commonMain/kotlin/io/github/kikin81/atproto/runtime/AtUriParser.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    if (segments.isEmpty() || segments.size > 3) return null
+    if (segments.any { it.isEmpty() }) return null
+
+    return AtUriParts(
+        repo = AtIdentifier(segments[0]),
+        collection = segments.getOrNull(1)?.let(::Nsid),
+        rkey = segments.getOrNull(2)?.let(::RecordKey),
+        fragment = fragment,
+    )
```
[the remark this answers is not in the document]
Comment (kikin81) [reply runtime/src/commonMain/kotlin/io/github/kikin81/atproto/runtime/AtUriParser.kt:72]: Not taking this one. The wrapper constructors (`AtIdentifier`, `Nsid`, `RecordKey`) are bare `value class` over `String` today and cannot throw, so the proposed try/catch is purely speculative scaffolding for a hypothetical future change.

Two reasons to leave it:

1. The project has a stated rule against adding error handling for scenarios that can't happen ("Don't add error handling, fallbacks, or validation for scenarios that can't happen. Trust internal code and framework guarantees.").
2. When the wrapper-validation ticket lands, *that* PR should make the deliberate decision about whether validation failures should null-out the whole parse or surface per-segment (likely the latter — callers usually want to know which segment was bad). Pre-deciding "swallow everything" here forecloses that design.

Today's `parseOrNull()` contract — "never throws on structural failure" — is honest and accurate. Will revisit when the validators actually exist.

File: runtime/src/commonTest/kotlin/io/github/kikin81/atproto/runtime/AtUriParserTest.kt [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        val parts = AtUri("at://did:plc:abc/app.bsky.feed.post").parse()
+        assertEquals(AtIdentifier("did:plc:abc"), parts.repo)
+        assertEquals(Nsid("app.bsky.feed.post"), parts.collection)
+        assertNull(parts.rkey)
+        assertNull(parts.fragment)
+    }
+
+    @Test
+    fun parse_trailingSlashAfterRepo_normalizesToRepoOnly() {
+        val parts = AtUri("at://did:plc:abc/").parse()
```
[the remark this answers is not in the document]
Comment (kikin81) [reply runtime/src/commonTest/kotlin/io/github/kikin81/atproto/runtime/AtUriParserTest.kt:48]: Added in 2b3bde1f:

- `parse_trailingSlashAfterRkey_throws` — `at://repo/coll/rkey/` now rejects (per the comment-1 tightening).
- `parse_doubleTrailingSlashAfterCollection_throws` — `at://repo/coll//` rejects (was already the behavior, now locked in).

File: runtime/src/commonTest/kotlin/io/github/kikin81/atproto/runtime/AtUriParserTest.kt [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+    @Test
+    fun parse_emptyPathSegment_throws() {
+        assertFailsWith<IllegalArgumentException> {
+            AtUri("at://did:plc:abc//rkey").parse()
+        }
+    }
+
+    @Test
+    fun parse_emptyCollectionWithTrailingRkey_throws() {
```
[the remark this answers is not in the document]
Comment (kikin81) [reply runtime/src/commonTest/kotlin/io/github/kikin81/atproto/runtime/AtUriParserTest.kt]: Renamed to `parse_doubleTrailingSlash_throws` in 2b3bde1f.
